lib/minitest/snapshots/assertion_extensions.rb

```diff
@@ -1,22 +1,21 @@
+require_relative "serializer"
 require "fileutils"
 
 module Minitest
   module Assertions
-    def assert_matches_snapshot(string, snapshot_name = nil)
-      snapshot_file = snapshot_path(self.class.name.underscore, snapshot_name || (@snapshot_assertion_counter += 1))
+    def assert_matches_snapshot(value, snapshot_name = nil)
+      snapshot_file = snapshot_path(self.class.name, snapshot_name || (@snapshot_assertion_counter += 1))
+      snapshot = Minitest::Snapshots::Serializer.serialize(value)
 
       if !Minitest::Snapshots.force_updates && File.exist?(snapshot_file)
-        assert_equal File.read(snapshot_file), string, "The string does not match the snapshot (located at #{snapshot_file})"
+        assert_equal File.read(snapshot_file), snapshot, "The value does not match the snapshot (located at #{snapshot_file})"
       else
         if Minitest::Snapshots.lock_snapshots
-          assert false, "New snapshot tried to be created when snapshot writing is prevented with the --lock-snapshots option."
+          assert false, "Attempt to create a snapshot when writing is prevented by the --lock-snapshots option."
```

Comment:
Better message but i might put a verb in it so that it reads without needing to understand the context like `Attempt to create a snapshot failed because writing is prevented by the --lock-snapshots option`

Reply:
Done.